src/mlpack/methods/emst/dtb_impl.hpp

```diff
@@ -103,6 +103,56 @@ template<
     template<typename TreeMetricType,
              typename TreeStatType,
              typename TreeMatType> class TreeType>
+DualTreeBoruvka<MetricType, MatType, TreeType>::DualTreeBoruvka(
+    Tree& tree,
+    const MetricType metric) :
+    tree(new Tree(tree)),
+    data(&this->tree->Dataset()),
+    ownTree(true),
+    naive(false),
+    connections(data.n_cols),
+    totalDist(0.0),
+    metric(metric)
```

Comment:
Hi @rcurtin ,

```
error: invalid initialization of reference of type ‘const arma::Mat<double>&’ from expression of type ‘arma::Mat<double>*’
     metric(metric)
                  ^
```
The `NeighborSearch` class is implemented in a similar way, but that runs without an error.  Any thoughts?